src/test/scala/com/raphtory/algorithms/WeightedGraphBuilder.scala

```diff
@@ -0,0 +1,30 @@
+package com.raphtory.algorithms
+
+import com.raphtory.core.components.graphbuilder.GraphBuilder
+import com.raphtory.core.components.graphbuilder.ImmutableProperty
+import com.raphtory.core.components.graphbuilder.Properties
+import com.raphtory.core.components.graphbuilder.LongProperty
+
+class WeightedGraphBuilder extends GraphBuilder[String] {
+
+  override def parseTuple(tuple: String): Unit = {
+    val line       = tuple.split(",")
+    val sourceNode = line(0)
+    val srcID      = sourceNode.toLong
+    val targetNode = line(1)
+    val tarID      = targetNode.toLong
+    val timeStamp  = line(2).toLong
+    val weight     = line(3).toLong
```

Comment:
Good call adding a weighted graph builder! Just a thought, I remember seeing something in a networks library (possibly graph-tool?) for reading in weighted networks from csv where the user could specify which columns in their csv correspond to `source`, `target`, `weight` defaulting to 0, 1, 2, respectively, in case the csv is not already in this format. Might be an idea to do something similar here, possibly also a separator defaulting to `,` to allow for tab separated files?

Reply:
Making a generic EdgeListGraphBuilder which would support these things has been in the back of my mind for a while but haven't had a chance to work on it yet. This was just a quick one for testing everything works.